papers/plutus-platform/authoring.tex

```diff
@@ -0,0 +1,307 @@
+The idea is to wrap the target Haskell code inside a splice of a Template Haskell function that inserts a marker around that AST fragment.
+
+This function does not actually compile the AST of the target program fragment.
+Instead, it inserts a marker function that is picked up by the \gls{plutus-tx} compiler injected with the plugin.
+When the plugin runs, it finds the marker, compiles the code, and inserts the serialized form back into the program.
+By taking a little care over the types of our marker function, we can ensure that the expression in question remains well-typed at each stage of this process.
+
+\subsubsection{Compiling GHC Core to Plutus Core}
+
+Both GHC Core and \gls{plutus-core} are extensions of \gls{system-f}.
+GHC Core is a much more generous extension.
+It adds mutually-recursive binding groups, algebraic data types, case expressions, coercions, and more.
+In contrast, \gls{plutus-core}, as discussed in \cref{sec:plutus-core}, is much more minimal.
+
+How do we deal with the extra features of GHC Core?
+First, we split the problem in half, by defining an intermediate language, \gls{plutus-ir} (see \cref{sec:plutus-ir}), which is much closer to GHC Core.
+Most of the theoretical complexity is therefore moved to the \gls{plutus-ir} compiler.
+
+The remaining work of the \gls{plutus-tx} compiler is then to \emph{lower} the GHC Core terms and types into their corresponding \gls{plutus-ir} variants, emitting errors as appropriate if we encounter features we do not support.
+
+\subsubsection{Supporting Haskell's features}
+
+As alluded to in the previous section, we do not support the entirety of Haskell.
+Thanks to the design of GHC, we get a great deal for free, as we compile programs after they have been converted to GHC Core, which means that most of the complex source-level features of Haskell have already been desugared into a smaller set of simpler features.
+
+While we support most ``standard'' Haskell, there are quite a few things we do not support. A non-exhaustive list of features that we do not support is:
+\begin{itemize}
+\item Not implemented yet
+  \begin{itemize}
+  \item Mutually recursive datatypes (should be done by release)
+  \end{itemize}
+\item Incompatible with the design of Plutus Core
+  \begin{itemize}
+  \item \code{PolyKinds}, \code{DataKinds}, anything that moves towards "Dependent Haskell"
+  \end{itemize}
+\item Technically difficult
+  \begin{itemize}
+  \item Literal patterns
+  \end{itemize}
+\item Requires access to function definitions (might be fixed with some GHC work)
+  \begin{itemize}
+  \item Function usage without \code{INLINEABLE} or \code{-fexpose-all-unfoldings}
+  \item Typeclass dictionaries
+  \end{itemize}
+\item Use of coercions required
+  \begin{itemize}
+  \item GADTs
+  \item \code{Data.Coerce}
+  \item \code{DerivingVia}, \code{GeneralizedNewtypeDeriving}, etc.
+  \end{itemize}
+\item Assumes "normal" codegen
```

Comment:
```suggestion
\item Assumes ``normal'' codegen
```

That, and it doesn't make any sense because the scripts are sandboxed and have no side effects other than burning CPU cycles